config/checkstyle-examples-checks.xml

```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
+        "https://checkstyle.org/dtds/configuration_1_3.dtd">
+
+<module name="Checker">
+  <property name="charset" value="UTF-8"/>
+  <property name="haltOnException" value="false"/>
+
+  <property name="cacheFile" value="target/cache_xdocs_examples"/>
+
+  <module name="SuppressionFilter">
+    <property name="file" value="${checkstyle.examples-suppressions.file}"/>
+  </module>
+
+  <module name="FileTabCharacter">
+    <property name="eachLine" value="false"/>
+  </module>
+
+  <module name="FileLength">
+    <property name="max" value="45"/>
+  </module>
+
+  <module name="RegexpOnFilename">
+    <property name="id" value="exampleFileName"/>
+    <property name="fileNamePattern" value="^Example\d+"/>
+    <property name="ignoreFileNameExtensions" value="true"/>
+    <property name="match" value="false"/>
+    <message key="regexp.filename.mismatch" value="All example files should start with Example\d+"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="noTrailingWhitespace"/>
+    <property name="format" value="\s+$"/>
+    <property name="minimum" value="0"/>
+    <property name="maximum" value="0"/>
+    <property name="message" value="Trailing whitespace is not allowed"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="noDefaultProperties"/>
+    <property name="format" value=".+\s=\s\(default\).*"/>
+    <property name="message" value="(default) properties are not allowed"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="xdocStartComment"/>
+    <property name="format" value="// xdoc section \-\- start"/>
+    <property name="message" value="Missing comment '// xdoc section -- start'"/>
+    <property name="minimum" value="1"/>
+    <property name="maximum" value="1"/>
+  </module>
+
+  <module name="RegexpSingleline">
+    <property name="id" value="xdocEndComment"/>
+    <property name="format" value="// xdoc section \-\- end"/>
+    <property name="message" value="Missing comment '// xdoc section -- end'"/>
+    <property name="minimum" value="1"/>
+    <property name="maximum" value="1"/>
+  </module>
+
+  <module name="LineLength">
+    <property name="max" value="85"/>
+  </module>
+
+  <module name="TreeWalker">
+    <module name="Indentation">
+      <property name="basicOffset" value="2"/>
+      <property name="caseIndent" value="2"/>
+      <property name="throwsIndent" value="2"/>
+      <property name="arrayInitIndent" value="2"/>
+      <property name="lineWrappingIndentation" value="2"/>
+    </module>
+
+    <module name="RegexpSinglelineJava">
+      <property name="id" value="exampleClassName"/>
+      <property name="format" value="^class Example\d+"/>
```

Comment:
Convert to `TypeName`.

Reply:
Not sure about this one. `TypeName` is not very flexible in what it's checking. I tried the following
```
    <module name="TypeName">
      <property name="format" value="Example\d+"/>
      <property name="tokens" value="CLASS_DEF"/>
      <property name="applyToPrivate" value="false"/>
      <property name="applyToPublic" value="false"/>
      <property name="applyToProtected" value="false"/>
    </module>
```
```
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:11:18: Name 'AbstractFirst' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:12:18: Name 'Second' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:13:9: Name 'AbstractThird' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:14:9: Name 'Fourth' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:15:18: Name 'GeneratorFifth' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:17:9: Name 'GeneratorSixth' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example2.java:12:18: Name 'AbstractFirst' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example2.java:13:18: Name 'Second' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example2.java:14:9: Name 'AbstractThird' must match pattern 'Example\d+'. [TypeName]
....
```

https://checkstyle.sourceforge.io/config_naming.html#TypeName

Reply:
@stoyanK7 Please pick one example and provide a link to it why you disagree with violation. Are these sub-classes that are the issue?

Reply:
Apologies for not having enough info.

Let's take this example. Yes, it has subclasses.
https://github.com/checkstyle/checkstyle/blob/fbbf38a0d9e3517cbea6e41ab95293d6ae20bbb6/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java#L10-L19

```
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:11:18: Name 'AbstractFirst' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:12:18: Name 'Second' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:13:9: Name 'AbstractThird' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:14:9: Name 'Fourth' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:15:18: Name 'GeneratorFifth' must match pattern 'Example\d+'. [TypeName]
[ERROR] [checkstyle] [ERROR] /home/stoyan/open-source/checkstyle/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java:17:9: Name 'GeneratorSixth' must match pattern 'Example\d+'. [TypeName]

```


I disagree because it cannot be configured to exclude those internal ones. We want to look for a top-level class matching the pattern. If `TypeName` can be configured to do just this, I am fine.

Reply:
Thanks, it looks like this is a limitation of the check. We can ignore for now.

@romani Does it make sense to allow this check to be configured between top level and sub-classes?

Reply:
In real life codding, I do not see reason to differentiate rules for top level and nested.

If any need appear, user should use xpath suppression to avoid unwanted violations.
We can use this Check with xpath Suppression. But let's do this in separate issue

Reply:
Created https://github.com/checkstyle/checkstyle/issues/13290